src/services/BlockchainScanner.ts

```diff
@@ -14,12 +17,19 @@ export type Result = {
 @injectable()
 export abstract class BlockchainScanner {
   @inject('Logger') logger!: Logger;
+  @inject('Settings') settings!: Settings;
+  @inject(BlockchainProviderRepository) blockchainProviderRepository!: BlockchainProviderRepository;
 
-  startBlock!: number;
-  blockchainId!: string;
+  startBlock: number;
   markerId!: string;
-  step!: number;
+  step: number;
   provider: StaticJsonRpcProvider | undefined;
+  chainId?: ChainsIds;
+
+  constructor() {
+    this.startBlock = 0;
```

Comment:
why we set this to 0? this should be set by settings. we never scan from 0.

Reply:
Removed